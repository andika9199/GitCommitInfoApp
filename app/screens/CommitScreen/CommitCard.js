import React from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';
import {
  WHITE,
  GRAY_DARKER,
  GRAY_LIGHT,
  TEXT_LARGE_BOLD,
  DEFAULT_TEXT,
} from '../../configs/style';

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 16,
    paddingVertical: 4,
    borderWidth: 1,
    borderColor: GRAY_LIGHT,
  },
  commitInfoContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    marginTop: 2,
  },
  imageContainer: {
    flexDirection: 'row',
  },
  image: {
    width: 20,
    height: 20,
    resizeMode: 'contain',
  },
  wording: {...DEFAULT_TEXT},
  bgGray: {
    backgroundColor: GRAY_LIGHT,
  },
  bgWhite: {
    backgroundColor: WHITE,
  },
});
export default function CommitCard({props}) {
  const {
    message,
    date,
    authorAvatar,
    authorName,
    commiterAvatar,
    commiterName,
    index,
  } = props;
  const isAuthorNotExist = !authorName || !authorAvatar ? true : false;
  const wording =
    (!isAuthorNotExist ? `${authorName} authored and ` : '') +
    `${commiterName} commited ${date}`;
  const color = index % 2 === 0 ? styles.bgGray : styles.bgWhite;
  return (
    <View style={[styles.container, color]}>
      <Text numberOfLines={1} style={TEXT_LARGE_BOLD}>
        {message}
      </Text>
      <View style={styles.commitInfoContainer}>
        <View style={styles.imageContainer}>
          {!isAuthorNotExist && (
            <Image source={{uri: authorAvatar}} style={styles.image} />
          )}
          <Image source={{uri: commiterAvatar}} style={styles.image} />
        </View>
        <View style={{flex: 1, paddingLeft: 4}}>
          <Text style={styles.wording}>{wording}</Text>
        </View>
      </View>
    </View>
  );
}
