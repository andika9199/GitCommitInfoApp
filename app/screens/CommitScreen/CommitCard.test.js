import React from 'react';
import renderer from 'react-test-renderer';
import CommitCard from './CommitCard';

const mockProps = {
  message: 'Always return an EventDispatcher in bridgeless mode...',
  date: '2020-05-27T19:16:21Z',
  authorAvatar: 'https://avatars1.githubusercontent.com/u/2308395?v=4',
  authorName: 'ejanzer',
  commiterAvatar: 'https://avatars3.githubusercontent.com/u/6422482?v=4',
  commiterName: 'facebook-github-bot',
  index: 1,
};

const mockNoAuthor = {
  message: 'Always return an EventDispatcher in bridgeless mode...',
  date: '2020-05-27T19:16:21Z',
  authorAvatar: '',
  authorName: '',
  commiterAvatar: 'https://avatars3.githubusercontent.com/u/6422482?v=4',
  commiterName: 'facebook-github-bot',
  index: 2,
};

it('CommitCard successfully rendered with data', () => {
  const component = renderer.create(<CommitCard props={mockProps} />);
  expect(component).toMatchSnapshot();
});

it('CommitCard successfully rendered without Author data', () => {
  const component = renderer.create(<CommitCard props={mockNoAuthor} />);
  expect(component).toMatchSnapshot();
});
