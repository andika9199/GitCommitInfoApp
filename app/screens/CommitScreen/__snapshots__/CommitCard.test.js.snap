// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitCard successfully rendered with data 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#cccccc",
        "borderWidth": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 4,
      },
      Object {
        "backgroundColor": "#ffffff",
      },
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#333333",
        "fontFamily": "Robot",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Always return an EventDispatcher in bridgeless mode...
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginTop": 2,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://avatars1.githubusercontent.com/u/2308395?v=4",
          }
        }
        style={
          Object {
            "height": 20,
            "resizeMode": "contain",
            "width": 20,
          }
        }
      />
      <Image
        source={
          Object {
            "uri": "https://avatars3.githubusercontent.com/u/6422482?v=4",
          }
        }
        style={
          Object {
            "height": 20,
            "resizeMode": "contain",
            "width": 20,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingLeft": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "fontFamily": "Robot",
            "fontSize": 13,
          }
        }
      >
        ejanzer authored and facebook-github-bot commited 2020-05-27T19:16:21Z
      </Text>
    </View>
  </View>
</View>
`;

exports[`CommitCard successfully rendered without Author data 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#cccccc",
        "borderWidth": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 4,
      },
      Object {
        "backgroundColor": "#cccccc",
      },
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#333333",
        "fontFamily": "Robot",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Always return an EventDispatcher in bridgeless mode...
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginTop": 2,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://avatars3.githubusercontent.com/u/6422482?v=4",
          }
        }
        style={
          Object {
            "height": 20,
            "resizeMode": "contain",
            "width": 20,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingLeft": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "fontFamily": "Robot",
            "fontSize": 13,
          }
        }
      >
        facebook-github-bot commited 2020-05-27T19:16:21Z
      </Text>
    </View>
  </View>
</View>
`;
